src/Engine/EmitMSIL/BuiltIn.cs

```diff
@@ -0,0 +1,383 @@
+﻿using ProtoCore.DSASM;
+using ProtoCore.Utils;
+using System;
+using System.Reflection;
+
+namespace EmitMSIL
+{
+    public class BuiltIn
+    {
+        private static object ToBoolean(object a)
+        {
+            switch (a)
+            {
+                case bool ab:
+                    return ab;
+
+                case long al:
+                    return al != 0;
+
+                case int ai:
+                    return ai != 0;
+
+                case null:
+                    return false;
+
+                case double ad:
+                    return !Double.IsNaN(ad) && !ad.Equals(0.0);
+
+                case string str:
+                    return !string.IsNullOrEmpty(str);
+
+                case char ac:
+                    return ac != 0;
+            }
+
+            if (a is Pointer)
```

Comment:
What is `Pointer`? Is it a C# type?

Reply:
I think so. This is from the old implementation.

Reply:
Is is a 'class System.Reflection.Pointer`.